src/lib/mina/errors.ts

```diff
@@ -60,23 +60,26 @@ function invalidTransactionError(
 ): string {
   let errorMessages = [];
   let rawErrors = JSON.stringify(errors);
+  let n = transaction.accountUpdates.length;
+  let accountUpdateErrors = errors.slice(1, n + 1);
 
-  // handle errors for fee payer
-  let errorsForFeePayer = errors[0];
-  for (let [error] of errorsForFeePayer) {
-    let message = ErrorHandlers[error as keyof typeof ErrorHandlers]?.({
-      transaction,
-      accountUpdateIndex: NaN,
-      isFeePayer: true,
-      ...additionalContext,
-    });
-    if (message) errorMessages.push(message);
+  // Check if the number of errors match the number of account updates. If there are more, then the fee payer has an error.
+  // We do this check because the fee payer error is not included in network transaction errors and is always present (even if empty) in the local transaction errors.
+  if (accountUpdateErrors.length === n) {
+    let errorsForFeePayer = errors[0];
+    for (let [error] of errorsForFeePayer) {
+      let message = ErrorHandlers[error as keyof typeof ErrorHandlers]?.({
+        transaction,
+        accountUpdateIndex: NaN,
+        isFeePayer: true,
+        ...additionalContext,
+      });
+      if (message) errorMessages.push(message);
+    }
   }
 
-  // handle errors for each account update
-  let n = transaction.accountUpdates.length;
-  for (let i = 0; i < n; i++) {
-    let errorsForUpdate = errors[i + 1];
+  for (let i = 0; i < accountUpdateErrors.length; i++) {
+    let errorsForUpdate = accountUpdateErrors[i];
```

Comment:
I might be confused, but to me this seems to have a bug: you created
```ts
let accountUpdateErrors = errors.slice(1, n + 1);
```
so if there is no fee payer error, then you're ignoring the errors for the 0th account update?